ccm_web/server/src/api/api.service.ts

```diff
@@ -21,4 +29,44 @@ export class APIService {
       }
     }
   }
+
+  async getCourseName (userLoginId: string, courseId: number): Promise<string | null> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId)
+    let name = null
+    try {
+      const response = await requestor.get<Course>(`courses/${courseId}`)
+      const course = response.body
+      name = course.name
+    } catch (error) {
+      if (error instanceof HTTPError && error.response.statusCode !== HttpStatus.OK) {
+        logger.error(`Received unusual status code ${String(error.response.statusCode)}`)
+        logger.error(`Response body: ${JSON.stringify(error.response.body)}`)
+      } else {
+        logger.error(`An error occurred while making a request to Canvas: ${JSON.stringify(error)}`)
+      }
+      throw new Error(error)
```

Comment:
when a Non `2xx` case do occur then should you be sending the error status code and message to FE instead of generalizing it. Since it would a value piece of info why the failure has occurred to user. When I tested this from Swagger this is what I see, I don't error is return back to controller.
**Error message which is valuable to send**
![Screen Shot 2021-06-15 at 4 49 38 PM](https://user-images.githubusercontent.com/8579775/122122031-4a7a9e00-cdfa-11eb-9224-f00bd963c8dd.png)

**Swagger console getting the error**
![Screen Shot 2021-06-15 at 4 52 57 PM](https://user-images.githubusercontent.com/8579775/122122107-5cf4d780-cdfa-11eb-84e4-f63e107d161f.png)

@chrisrrowland what do you preference,  since you might want to deal with API failures and message. So you have better vision on how this should be done

Reply:
I could do this, I guess I was trying to keep the error handling minimal for this first pass. But I could see returning an error object instead of `null` from the `APIService` functions, something like this:

```
{
  statusCode: 401,
  message: 'You are not authorized to modify this resource."
}

Reply:
The status code might be sufficient?  I'm not sure exactly what use cases there will be for telling the user what they should do when an error occurs.  If there is any error we can predict happening, it seems like we should have a better way to handle it than telling the user to act in some way.

However the actual translation of 401 to 'You are not authorized to modify this resource.' does seem like something that would come from the front end, as that is where any localizing would happen.. I think?  (Not that we are doing that)
I haven't actually worked on a web application with localized text so I'm not sure.

Reply:
Okay, as requested, I'm passing through Canvas error codes and messages now. I've added some parsing code for the Canvas error body shape (which doesn't seem to be well documented anywhere :( )